doc/prediction.rst

```diff
@@ -0,0 +1,142 @@
+``Ture`` to indicate a more restricted output is desired.  Assuming you are using
+:py:obj:`xgboost.Booster`, here is a list of possible returns:
+
+- When using normal prediction with ``strict_shape`` set to ``True``:
+
+  Output is a 2-dim array with first dimension as rows and second as groups.  For
+  regression/survival/ranking/binary classification this is equivalent to a column vector
+  with ``shape[1] == 1``.  But for multi-class with ``multi:softprob`` the number of
+  columns equals to number of classes.  If strict_shape is set to False then XGBoost might
+  output 1 or 2 dim array.
+
+- When using ``output_margin`` to avoid transformation and ``strict_shape`` is set to ``True``:
+
+  Similar to the previous case, output is a 2-dim array, except for that ``multi:softmax``
+  has equivalent output of ``multi:softprob`` due to dropped transformation.  If strict
+  shape is set to False then output can have 1 or 2 dim depending on used model.
+
+- When using ``preds_contribs`` with ``strict_shape`` set to ``True``:
+
+  Output is a 3-dim array, with ``(rows, groups, columns + 1)`` as shape.  Whether
+  ``approx_contribs`` is used does not change the output shape. If the strict shape
+  parameter is not set, it can be a 2 or 3 dimension array depending on whether
+  multi-class model is being used.
+
+- When using ``preds_interactions`` with ``strict_shape`` set to ``True``:
+
+  Output is a 4-dim array, with ``(rows, groups, columns + 1, columns + 1)`` as shape.
+  Like the predict contribution case, whether ``approx_contribs`` is used does not change
+  the output shape.  If strict shape is set to False, it can have 3 or 4 dims depending on
+  the underlying model.
+
+- When using ``pred_leaf`` with ``strict_shape`` set to ``True``:
+
+  Output is a 4-dim array with ``(n_samples, n_iterations, n_classes, n_trees_in_forest)``
+  as shape.  ``n_trees_in_forest`` is specified by the ``numb_parallel_tree`` during
+  training.  When strict shape is set to False, output is a 2-dim array with last 3 dims
+  concatenated into 1.  When using ``apply`` method in scikit learn interface, this is set
+  to False by default.
+
+
+Other than these prediction types, there's also a parameter called ``iteration_range``,
+which is similar to model slicing.  But instead of actually splitting up the model into
+multiple stacks, it simply returns the prediction formed by the trees within range.
+
+
+*********
+Predictor
+*********
+
+There are 2 predictors in XGBoost (3 if you have the one-api plugin enabled), namely
+``cpu_predictor`` and ``gpu_predictor``.  The default option is ``auto`` so that XGBoost
+can employ some heuristics for saving GPU memory during training.  They might have slight
+different outputs due to floating point errors.
+
+
+***********
+Base Margin
+***********
+
+There's a parameter in XGBoost called ``base_score``, and a meta data called
+``base_margin``.  They specifies the global bias for boosted model.  If the latter is
```

Comment:
Clarify that base margin can be set as metadata on DMatrix, base score can be modified as a parameter.

Reply:
Done.